### ISSUE TITLE
[tree view] Keyboard navigation error - MUI RichTreeView

### ISSUE THREAD

**Issue by cybadave** [mui/mui-x#15649]
### Steps to reproduce

Steps:
1. Open this link to live example: [RichTreeViewTest](https://cybadave.github.io/RichTreeViewTest/)
2. Expand the first root node "1" by clicking on it
3. Expand the node "1 2" by clicking on it
4. Press the down arrow key until focus has moved to the second root node "2"
5. Press the up arrow key once
6. Observe that node "1 2 1" is focused

### Current behavior

Node "1 2 1" is focused

### Expected behavior

Node "1 2 5" should be focused

### Context

RichTreeViewNodes should be consistently accessible via keyboard navigation

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

  System:
    OS: Windows 11 10.0.22621
  Binaries:
    Node: 20.16.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.8.1 - C:\Program Files\nodejs\npm.CMD
    pnpm: Not Found
  Browsers:
    Chrome: Not Found
    Edge: Chromium (127.0.2651.74)
  npmPackages:
    @emotion/react: ^11.13.5 => 11.13.5
    @emotion/styled: ^11.13.5 => 11.13.5
    @mui/core-downloads-tracker:  6.1.8
    @mui/material: ^6.1.8 => 6.1.8
    @mui/private-theming:  6.1.8
    @mui/styled-engine:  6.1.8
    @mui/system:  6.1.8
    @mui/types:  7.2.19
    @mui/utils:  6.1.8
    @mui/x-internals:  7.21.0
    @mui/x-tree-view: ^7.22.1 => 7.22.1
    @types/react: ^18.3.12 => 18.3.12
    react: ^18.3.1 => 18.3.1
    react-dom: ^18.3.1 => 18.3.1
    typescript: ^4.9.5 => 4.9.5

Browsers:

Firefox 131.0 (64Bit)
Microsoft Edge for Business 131.0.2903.63 (Official build) (64-bit)
Google Chrome 131.0.6778.86 (Official Build) (64-bit)

**Search keywords**: MUI RichTreeView keyboard navigation

**Comment by flaviendelangle** [comment]
The keyboard navigation is in the `useTreeViewKeyboardNavigation` hook, I suspect that the issue is in the `getPreviousNavigableItem` method.
